src/shared/jwt.js

```diff
@@ -0,0 +1,90 @@
+const jwt = require('jsonwebtoken');
+const config = require('../config');
+
+/**
+ * Validates user api JWT.
+ *
+ * If token is invalid (cannot be decoded): return status 500 and error
+ * If token is not in header: return status 401
+ */
+let validateUserToken = (req, res, next) => {
+    let token = req.headers['ha-api-token'];
+
+    if (token) {
+        jwt.verify(token, config.userPublicKey, config.verifyOptions, (err) => {
+            if (err)
+                res.status(500).send(err);
+
+            next();
+        });
+    }
+
+    return res.status(401).send();
+};
+
+/**
+ * Validates admin api JWT.
+ *
+ * If token is invalid (cannot be decoded): return status 500 and error
+ * If token is not in header: return status 401
+ */
+let validateAdminToken = (req, res, next) => {
+    let token = req.headers['ha-admin-token'];
```

Comment:
I like the clean distinction here between the admin and regular tokens in the header.